pcdsdevices/ipm.py

```diff
@@ -49,15 +96,276 @@ def removed(self):
+    ch8 = FCpt(Wave8Channel, '{self.prefix}', channel_index=8, kind='normal')
+    ch9 = FCpt(Wave8Channel, '{self.prefix}', channel_index=9, kind='normal')
+    ch10 = FCpt(Wave8Channel, '{self.prefix}', channel_index=10, kind='normal')
+    ch11 = FCpt(Wave8Channel, '{self.prefix}', channel_index=11, kind='normal')
+    ch12 = FCpt(Wave8Channel, '{self.prefix}', channel_index=12, kind='normal')
+    ch13 = FCpt(Wave8Channel, '{self.prefix}', channel_index=13, kind='normal')
+    ch14 = FCpt(Wave8Channel, '{self.prefix}', channel_index=14, kind='normal')
+    ch15 = FCpt(Wave8Channel, '{self.prefix}', channel_index=15, kind='normal')
+
+    def __init__(self, prefix, *, name, prefix_ioc=None, **kwargs):
+        if not prefix_ioc:
+            self._prefix_ioc = 'IOC:%s' % prefix
+        else:
+            self._prefix_ioc = prefix_ioc
+        super().__init__(prefix, name=name, **kwargs)
+
+    def screen(self):
+        """
+        Function to call the (pyQT) screen for a wave8 box
+        """
+        executable = '/reg/g/pcds/pyps/apps/wave8/latest/wave8'
+        if shutil.which(executable) is None:
+            print('%s is not on path, we cannot start the screen' % executable)
+            return
+        os.system('%s --base %s --ioc %s --evr %s &' %
+                  (executable, self.prefix, self._prefix_ioc,
+                   self.prefix+':TRIG'))
+
+
+class IPM_IPIMB(IPMMotion):
+    """
+%s
+
+    has a member ipimb which represents the ipimb box used for readout
+    """
+    __doc__ = __doc__ % (IPM_base) + basic_positioner_init
+
+    ipimb = FCpt(IPIMB, '{self.prefix_ipimb}', prefix_ioc='{self.prefix_ioc}')
+
+    def __init__(self, prefix, *, name, **kwargs):
+        self.prefix_ipimb = kwargs.pop('prefix_ipimb')
+        self.prefix_ioc = kwargs.pop('prefix_ioc', None)
+        super().__init__(prefix, name=name, **kwargs)
+
+    def isum(self):
+        return self.ipimb.isum.get()
+
+    def xpos(self):
+        return self.ipimb.xpos.get()
+
+    def ypos(self):
+        return self.ipimb.ypos.get()
+
+    def channel(self, i=0):
+        # thrown an exception for an invalid channel?
+        # if ( i >= 4 ): return None
+        return getattr(self.ipimb, 'ch%i').amplitude.get()
+
+    def channels(self):
+        return [getattr(self.ipimb, 'ch%i').amplitude.get() for i in range(4)]
+
+
+class IPM_Wave8(IPMMotion):
+    """
+%s
+
+    has a member wave8 which represents the wave8 used for readout
+    """
+    __doc__ = __doc__ % (IPM_base) + basic_positioner_init
+
+    wave8 = FCpt(Wave8, '{self.prefix_wave8}')
+
+    def __init__(self, prefix, *, name, **kwargs):
+        self.prefix_wave8 = kwargs.pop('prefix_wave8')
+        super().__init__(prefix, name=name, **kwargs)
+
+    def isum(self):
+        return self.wave8.isum.get()
+
+    def xpos(self):
+        return self.wave8.xpos.get()
+
+    def ypos(self):
+        return self.wave8.ypos.get()
```

Comment:
I'm getting a bit of deja vu with the methods here. There's likely a concise way to not have to copy/paste code between these two classes. You don't have to find it now, we can refactor later if someone figures out how to do it.